modules/webgl/src/classes/framebuffer.js

```diff
@@ -23,12 +23,24 @@ export default class Framebuffer extends Resource {
     } = {}
   ) {
     let supported = true;
-    supported =
-      colorBufferFloat &&
-      gl.getExtension(isWebGL2(gl) ? 'EXT_color_buffer_float' : 'WEBGL.color_buffer_float');
-    supported =
-      colorBufferHalfFloat &&
-      gl.getExtension(isWebGL2(gl) ? 'EXT_color_buffer_float' : 'EXT_color_buffer_half_float');
+
+    if (colorBufferFloat) {
+      supported = Boolean(
+        gl.getExtension('EXT_color_buffer_float') ||
```

Comment:
Should we build this into our features system instead of creating "a shadow features system" in `Framebuffer.isSupported`?

Now the WebGL1/2 extensions are a complicated mess so I might miss something here, but I do not quite understand the logic: `OES_texture_float` represents the ability to read from floating point textures, whereas `WEBGL_color_buffer_float` and `EXT_color_buffer_float` represent the ability to write to float point textures.

I could see the value for a function that checked both read and write at one time (though we already provide `hasFeatures` for that).

However since this code uses `||` operator, the result will be true if the framebuffer either supports reading, or writing, or both, which does not seem that useful?

Reply:
OES_texture_float implicitly enables color_buffer_float, and is the only way it's reported in some implementations (e.g. Safari).

See the last line in the overview section here: https://www.khronos.org/registry/webgl/extensions/OES_texture_float/

Reply:
Ok thans. I'd personally really appreciate if a comment with that link was added to the source code.

Also, the test is so brief that I can barely understand what they are saying: that this extension implicitly activates the write extension if that one is present...  

But if not present it won't be activated so we still need to check for it?

Reply:
I believe in most cases where OES_texture_float is available you'll be able to render to float textures, but I have heard cases where that wasn't the case, mostly on older mobile devices. If we want to be 100% sure, we'd have to create a framebuffer with a float render target, try to draw to it, and check for errors. My feeling is this would be overkill, but I have seen some engines do it: https://github.com/playcanvas/engine/blob/9ab0893e6100473df41002094fdf18b7028c1081/src/graphics/device.js#L31

Reply:
I don't think we need to go that far. I would just appreciate a comment with a precise description of why we are doing it this way, e.g. 

```
// On Safari WebGL1, floating point textures are renderable when `OES_texture_float` is supported, even though the `WEBGL_colorbuffer_float` extension is not supported. See comment in standard: ...
```

Without that I don't think I could maintain this code (i.e. understand why it was written like this).

Reply:
OK I see the comments now, looks good, thanks for bearing with me.

Reply:
No problem. Thanks for keeping me honest 😄